app/controllers/users_controller.rb

```diff
@@ -37,6 +45,31 @@ def create
     end
   end
 
+  def create_director
+    @user= User.new(director_params)
+    @user.role = "director"
+    if @user.save
+      puts "USER SAVED"
+      @director = Director.new(
+        user_id: @user.id
+      )
+      if @director.save
+        puts "DIRECTOR SAVED"
+        UserMailer.registration_confirmation(@user).deliver
+        flash[:notice] = "Director Created! Please ask new director to verify email"
+        redirect_to director_dashboard_index_path
+      else
+        puts "DIRECTOR NOT SAVED"
+        flash[:alert] = "Oops! Something went wrong, please try again"
+        redirect_to new_director_director_path
+      end
+    else
+      puts "USER NOT SAVED"
+      flash[:alert] = "Oops! Something went wrong, please try again!"
+      redirect_to new_director_director_path
```

Comment:
Use the redirect_to_role_portal method, but update it so that it goes to the proper pages.

Also, it looks like if the director is saved, it goes to the director dashboard. It should go to the new_director_director_path page. 

Have you tested this? Please go through it and make intentional errors in the forms and also create the director/parents successfully. I have a feeling this won't work as you expect.